packages/pools/src/router/__tests__/routes.spec.ts

```diff
@@ -0,0 +1,412 @@
+import { Int } from "@keplr-wallet/unit";
+
+import {
+  allPools,
+  makeDefaultTestRouterParams,
+  makeStablePool,
+  makeWeightedPool,
+} from "./pool";
+
+describe("OptimizedRoutes", () => {
+  describe("Simple: picks higher liquidity routes", () => {
+    test("picks higher liquidity pool", async () => {
+      const pools = [
+        makeWeightedPool({
+          id: "2",
+          firstPoolAsset: { amount: "1000" },
+          secondPoolAsset: { amount: "1000" },
+        }),
+        makeWeightedPool(),
+      ];
+
+      const router = makeDefaultTestRouterParams({
+        pools,
+        incentivizedPoolIds: [],
+        stakeCurrencyMinDenom: "ufoo",
+      });
+
+      const routes = await router.getOptimizedRoutesByTokenIn(
+        {
+          denom: "uion",
+          amount: new Int("100"),
+        },
+        "uosmo"
+      );
+
+      expect(routes.length).toBe(1);
+      expect(routes[0].pools[0].id).toBe("1");
```

Comment:
this fails locally and picking pool `2` instead of `1`. perhaps a bug in `getOptimizedRoutesByTokenIn` when sorting routes by weight?